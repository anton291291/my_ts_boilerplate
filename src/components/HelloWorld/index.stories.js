import React from 'react';
import HelloWorld from './index';
import { withKnobs, text, number } from '@storybook/addon-knobs';

export default {
    title: '/HelloWorld',
    component: HelloWorld,
    decorators: [withKnobs]
};

export const HelloUser = () => <HelloWorld username={text('username', '')} />;
