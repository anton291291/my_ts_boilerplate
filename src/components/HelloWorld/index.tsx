import React from 'react';
import { StyledBtn } from '../StyledComponents';

type Props = { username?: string };

const HelloWorld: React.FC<Props> = (props) => {
    const { username } = props;

    const handleClick = () => {
        alert('Click!');
    };

    return <StyledBtn onClick={handleClick}>Hello World {username}</StyledBtn>;
};

export default HelloWorld;
