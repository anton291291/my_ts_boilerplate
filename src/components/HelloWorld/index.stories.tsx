import React from 'react';
import HelloWorld from './index';
import { withKnobs, text, number } from '@storybook/addon-knobs';

export default {
    title: '/HelloWorld',
    component: HelloWorld,
    decorators: [withKnobs]
};

export const HelloUser = () => (
    <HelloWorld
        width={number('Width', 100, {
            range: true,
            min: 50,
            max: 500,
            step: 1
        })}
        username={text('username', '')}
    />
);
