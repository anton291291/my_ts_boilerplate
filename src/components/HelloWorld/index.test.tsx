import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import HelloWorld from './index';

describe('HelloWorld', () => {
    test('should render correctly', () => {
        const output = shallow(<HelloWorld />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    test('should handle the click event', () => {
        window.alert = jest.fn();
        const output = shallow(<HelloWorld />);
        output.simulate('click');
        expect(window.alert).toHaveBeenCalledWith('Click!');
    });
});
