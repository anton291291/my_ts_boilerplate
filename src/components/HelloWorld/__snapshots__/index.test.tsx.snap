// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld should render correctly 1`] = `
<StyledBtn
  onClick={[Function]}
>
  Hello World 
</StyledBtn>
`;
